cmd/nerdctl/container_run_linux_test.go

```diff
@@ -150,6 +152,38 @@ func TestRunAddHost(t *testing.T) {
 		return nil
 	})
 	base.Cmd("run", "--rm", "--add-host", "10.0.0.1:testing.example.com", testutil.AlpineImage, "cat", "/etc/hosts").AssertFail()
+
+	response := "This is the expected response for --add-host special IP test."
+	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+		io.WriteString(w, response)
+	})
+	const hostPort = 8081
+	s := http.Server{Addr: fmt.Sprintf(":%d", hostPort), Handler: nil, ReadTimeout: 30 * time.Second}
+	go s.ListenAndServe()
+	defer s.Shutdown(context.Background())
+	base.Cmd("run", "--rm", "--add-host", "test:host-gateway", testutil.NginxAlpineImage, "curl", fmt.Sprintf("test:%d", hostPort)).AssertOutExactly(response)
+}
+
+func TestRunAddHostWithCustomHostGatewayIP(t *testing.T) {
+	// Not parallelizable (https://github.com/containerd/nerdctl/issues/1127)
+	base := testutil.NewBase(t)
+	testutil.DockerIncompatible(t)
```

Comment:
why is it docker incompatible?

Reply:
```
docker run --host-gateway-ip "192.168.5.2" --add-host=name:host-gateway -it run amazonlinux bash
unknown flag: --host-gateway-ip
See 'docker run --help'.
```

Docker has host-gateway-ip in [config](https://github.com/moby/moby/blob/master/cmd/dockerd/config.go#L39) but not in global flag. From my understanding, there is a convention in nerdctl that transferring all the config to global flag first before using them. So the global flag is docker incompatible.